plugin/validator_test.go

```diff
@@ -956,6 +956,889 @@ func TestValidator_SchemaAsJSON(t *testing.T) {
+				"header_name": "x-header",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid character in header_name",
+			config: `{
+				"header_name": "!header",
+				"tag": "test"
+			}`,
+			wantErr: true,
+			expectedErr: `{"config":{"header_name":"bad header name '!header', ` +
+				`allowed characters are A-Z, a-z, 0-9, '_', and '-'"}}`,
+		},
+		{
+			name: "valid timeout",
+			config: `{
+				"timeout": 10,
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid negative timeout",
+			config: `{
+				"timeout": -1,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"timeout":"value should be between 0 and 2147483646"}}`,
+		},
+		{
+			name: "invalid too big timeout",
+			config: `{
+				"timeout": 2147483647,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"timeout":"value should be between 0 and 2147483646"}}`,
+		},
+		{
+			name: "valid uuid",
+			config: `{
+				"uuid": "23e623b0-ca23-11ec-9d64-0242ac120002",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid uuid",
+			config: `{
+				"uuid": "abc",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"uuid":"expected a valid UUID"}}`,
+		},
+		{
+			name: "invalid empty uuid",
+			config: `{
+				"uuid": "",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"uuid":"expected a valid UUID"}}`,
+		},
+		{
+			name: "invalid not-string uuid",
+			config: `{
+				"uuid": -1,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"uuid":"expected a string"}}`,
+		},
+		{
+			name: "valid auto_timestamp_s",
+			config: `{
+				"auto_timestamp_s": 1234,
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "valid number auto_timestamp_s",
+			config: `{
+				"auto_timestamp_s": 1.234,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"auto_timestamp_s":"expected an integer"}}`,
+		},
+		{
+			name: "invalid string auto_timestamp_s",
+			config: `{
+				"auto_timestamp_s": "1234",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"auto_timestamp_s":"expected an integer"}}`,
+		},
+		{
+			name: "valid auto_timestamp_ms",
+			config: `{
+				"auto_timestamp_ms": 1.234,
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid auto_timestamp_ms",
+			config: `{
+				"auto_timestamp_ms": "1.234",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"auto_timestamp_ms":"expected a number"}}`,
+		},
+		{
+			name: "valid name",
+			config: `{
+				"name": "foo.-_~",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid name",
+			config: `{
+				"name": "foo.-_~!",
+				"tag": "test"
+			}`,
+			wantErr: true,
+			expectedErr: `{"config":{
+				"name":"invalid value 'foo.-_~!': it must only contain alphanumeric and '., -, _, ~' characters"
+				}}`,
+		},
+		{
+			name: "valid utf8_name",
+			config: `{
+				"utf8_name": "foo.-_~",
+				"tag": "test"
+			}`,
+		},
```

Comment:
Can we add a test here with an ascii character that is outside of the standard alphanumeric characters?

Reply:
https://github.com/Kong/goks/pull/14/commits/eaa5261a98b197eb0b9c6ba3fd5b11efdd7fae38